docker-for-aws/persistent-data-volumes.md

```diff
@@ -8,58 +8,252 @@ title: Docker for AWS persistent data volumes
+The following examples show how to create swarm services that require data
+persistence using the `--mount` flag and specifying Cloudstor as the volume
+driver.
 
-### Share the same volume between tasks:
+### Share the same volume among tasks using EFS
+
+In those regions where EFS is supported and EFS support is enabled during
+deployment of the Cloud Formation template, you can use `shared` Cloudstor
+volumes to share access to persistent data across all tasks in a swarm service
+running in multiple nodes, as in the following example:
 
 ```bash
-docker service create --replicas 5 --name ping1 \
-    --mount type=volume,volume-driver=cloudstor:aws,source=sharedvol1,destination=/shareddata \
-    alpine ping docker.com
+$ docker service create \
+  --replicas 5 \
+  --name ping1 \
+  --mount type=volume,volume-driver=cloudstor:aws,source=sharedvol1,destination=/shareddata \
+  alpine ping docker.com
 ```
 
-Here all replicas/tasks of the service `ping1` share the same persistent volume `sharedvol1` mounted at `/shareddata` path within the container. Docker Swarm takes care of interacting with the Cloudstor plugin to make sure the common backing store is mounted on all nodes in the swarm where tasks of the service are scheduled. Each task needs to make sure they don't write concurrently on the same file at the same time and cause corruptions since the volume is shared.
+All replicas/tasks of the service `ping1` share the same persistent volume
+`sharedvol1` mounted at `/shareddata` path within the container. Docker takes
+care of interacting with the Cloudstor plugin to ensure that EFS is mounted on
+all nodes in the swarm where service tasks are scheduled. Your application needs
+to be designed to ensure that tasks do not write concurrently on the same file
+at the same time, to protect against data corruption.
 
-With the above example, you can make sure that the volume is indeed shared by logging into one of the containers in one swarm node, writing to a file under `/shareddata/` and reading the file under `/shareddata/` from another container (in the same node or a different node).
+You can verify that the same volume is shared among all the tasks by logging
+into one of the task containers, writing a file under `/shareddata/`, and
+logging into another task container to verify that the file is available there
+as well.
 
-### Use a unique volume per task:
+The only option available for EFS is `perfmode`. You can set `perfmode` to
+`maxio` for high IO throughput:
 
 ```bash
 {% raw %}
-docker service create --replicas 5 --name ping2 \
-    --mount type=volume,volume-driver=cloudstor:aws,source={{.Service.Name}}-{{.Task.Slot}}-vol,destination=/mydata \
-    alpine ping docker.com
+$ docker service create \
+  --replicas 5 \
+  --name ping3 \
+  --mount type=volume,volume-driver=docker4x/cloudstor:aws-v{{ edition_version }},source={{.Service.Name}}-{{.Task.Slot}}-vol5,destination=/mydata,volume-opt=perfmode=maxio \
```

Comment:
Same as before - Since we alias the plugin to prevent upgrade failures, shouldn't this `volume-driver` remain `cloudstor:aws`?